pkg/api/types.go

```diff
@@ -1205,6 +1221,13 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// Local represents a directory with specific topology contraints
+type LocalVolumeSource struct {
+	// The full path to the volume on the node
+	// For alpha, this path must be a file-based volume
+	Path string
```

Comment:
This is not what has been agreed in the proposal:

> [some] addon DaemonSet can also discover block devices and creates corresponding PVs for them with the `block` field.  
>```
>kind: PersistentVolume
>...
>spec:
>      localStorage:
>         block:
>           device: /var/lib/kubelet/storage-raw-devices/foo
>```

@vishh, @thockin, how are you going to extend LocalVolumeSource with block devices in future? Overloading `Path` with both path to a mounted directory and a device name looks ugly to me, it has totally different semantics. In addition, you need to put at least `fsType` somewhere into the structure and it will look weird for already mounted directories where it does not make any sense.